### PR TITLE
fix register type

### PR DESCRIPTION
**Pull request by bluebill1049**
can you take a look if this fix the type?

### CODE REVIEW

**Comment by bluebill1049** [comment]
if it's the problem with return, maybe we just return `any` for now?

**Comment by bluebill1049** [comment]
also how do I reproduce the return issue? it passes the unit test :(

File: src/contextTypes.ts [hunk 1]
```diff
@@ -27,11 +27,15 @@ export interface FormContextValues<
...
   register<Element extends ElementLike = ElementLike>(
     validateRule: ValidationOptions,
-  ): (instance: Element | null) => void;
+  ): (ref: Element | null) => void;
   register<Element extends ElementLike = ElementLike>(
-    instance: Element,
+    ref: Element | null,
     validationOptions?: ValidationOptions,
   ): void;
+  register<Element extends ElementLike = ElementLike>(
```
Comment (stramel) [inline src/contextTypes.ts]: This declaration isn't needed I believe

File: src/useForm.ts [hunk 2]
```diff
@@ -524,13 +524,13 @@ export default function useForm<
...
+    ref: Element | null,
     validationOptions?: ValidationOptions,
-  ): undefined;
+  ): void;
   function register<Element extends ElementLike = ElementLike>(
-    refOrValidateRule: ValidationOptions | Element,
+    refOrValidateRule: ValidationOptions | Element | null,
     validationOptions?: ValidationOptions,
-  ): ((ref: Element | null) => void) | undefined {
+  ): ((ref: Element | null) => void) | void {
```
Comment (stramel) [inline src/useForm.ts]: I thought this gave me an error when I had it like this

**Pull request merged**